[CI] LossySumDoubleGroupingAggregatorFunctionTests testManyInitialManyPartialFinalRunner failing

**Build Scans:**
- [elasticsearch-intake #27703 / part3](https://gradle-enterprise.elastic.co/s/zixyehjdjdvpe)
- [elasticsearch-pull-request #88837 / part-3](https://gradle-enterprise.elastic.co/s/4dfe6jb45evcm)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:compute:test" --tests "org.elasticsearch.compute.aggregation.LossySumDoubleGroupingAggregatorFunctionTests.testManyInitialManyPartialFinalRunner" -Dtests.seed=E483DC755A39E4F3 -Dtests.locale=ru-UA -Dtests.timezone=America/Araguaina -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.compute.aggregation.LossySumDoubleGroupingAggregatorFunctionTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testManyInitialManyPartialFinalRunner'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a numeric value within <0.01> of <2537.308557464561>
     but: <2537.128132229189> differed by <0.1704252353722404> more than delta <0.01>
```

**Issue Reasons:**
- [main] 2 failures in test testManyInitialManyPartialFinalRunner (0.7% fail rate in 292 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -543,15 +543,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
   method: test
   issue: https://github.com/elastic/elasticsearch/issues/133077
-- class: org.elasticsearch.compute.aggregation.LossySumDoubleGroupingAggregatorFunctionTests
-  method: testManyInitialManyPartialFinalRunner
-  issue: https://github.com/elastic/elasticsearch/issues/133809
-- class: org.elasticsearch.compute.aggregation.LossySumDoubleGroupingAggregatorFunctionTests
-  method: testInitialIntermediateFinal
-  issue: https://github.com/elastic/elasticsearch/issues/133822
-- class: org.elasticsearch.compute.aggregation.LossySumDoubleGroupingAggregatorFunctionTests
-  method: testInitialFinal
-  issue: https://github.com/elastic/elasticsearch/issues/133829
 - class: org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT
   method: testGroupBySubset
   issue: https://github.com/elastic/elasticsearch/issues/133220

--- a/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/LossySumDoubleAggregator.java
+++ b/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/LossySumDoubleAggregator.java
@@ -144,8 +144,8 @@ class LossySumDoubleAggregator {
             public void add(int positionOffset, IntVector groupIds) {
                 if (groupIds.isConstant()) {
                     double sum = 0.0;
-                    int positionCount = groupIds.getPositionCount();
-                    for (int i = 0; i < positionCount; i++) {
+                    final int to = positionOffset + groupIds.getPositionCount();
+                    for (int i = positionOffset; i < to; i++) {
                         sum += values.getDouble(i);
                     }
                     state.add(sum, groupIds.getInt(0));

